invertedindex/core_index.go

```diff
@@ -0,0 +1,123 @@
+package invertedindex
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/filecoin-project/go-indexer-core"
+
+	"github.com/ipfs/go-cid"
+	"github.com/multiformats/go-multihash"
+
+	"github.com/filecoin-project/dagstore/shard"
+)
+
+var ErrNotFound = errors.New("cid not found in Index")
+
+var _ Index = (*IndexerCoreIndex)(nil)
+
+type IndexerCoreIndex struct {
+	is indexer.Interface
+}
+
+func NewIndexerCore(is indexer.Interface) *IndexerCoreIndex {
+	return &IndexerCoreIndex{
+		is: is,
+	}
+}
+
+func (d *IndexerCoreIndex) AddMultihashesForShard(mhIter MultihashIterator, s shard.Key) error {
+	return mhIter.ForEach(func(mh multihash.Multihash) error {
+		// go-indexer-core appends values to the existing values we already have for the key
+		// it also takes care of de-duplicating values.
+		_, err := d.is.Put(mh, valueForShardKey(s))
+		return err
+	})
+}
+
+func (d *IndexerCoreIndex) DeleteMultihashesForShard(sk shard.Key, mhIter MultihashIterator) error {
+	return mhIter.ForEach(func(mh multihash.Multihash) error {
+		// remove the given value i.e. shard key from the index for the given cid key.
+		_, err := d.is.Remove(mh, valueForShardKey(sk))
+		return err
+	})
+}
+
+func (d *IndexerCoreIndex) GetShardsForCid(c cid.Cid) ([]shard.Key, error) {
```

Comment:
I agree with Anton's suggestion that for this layer we should be consistent in only having multihashes in the interface, so this function would be
`GetShardsForMultihash(mh multihash.Multihash) ([]shard.Key, error)`